u1-request-response/app.js

```diff
@@ -0,0 +1,161 @@
+//Import libraries 
+const express = require('express')
+
+//Setup app 
+const app = express()
+
+//Middleware 
+app.use(express.static('public'))
+
+//localhost:3000
+app.get('/', (req, res) => {
+    res.send('better world')
+  })
+
+//localhost:3000/html
+app.get('/html', (req, res) => {
+    res.send(`
+    <html><body>
+        <h1>Technologies</h1>
+            <ul>
+                <li>JavaScript</li>
+                <li>Node</li>
+                <li>Express</li>
+            </ul>
+    </body></html>
+  `)
+  })
+
+//localhost:3000/Json
+  app.get('/json', (req, res) => {
+    res.json(
+        {
+            technologies: [
+                'javascript',`typescript`, 'Noodle', 'Express'
+            ]
+        })
+  })
+
+//localhost:3000/status
+  app.get(`/status` ,(req,res) => {
+      res.sendStatus(403)
+  })
+
+  app.get(`/status-200` ,(req,res) => {
+    res.sendStatus(200)
+})
+
+app.get(`/status-400` ,(req,res) => {
+    res.sendStatus(400)
+})
+
+    //localhost:3000/status2
+    app.get(`/status2` ,(req,res) => {
+        res.status(500).send(`Oh no,something went wrong`)
+    })
+
+//locahost:3000/html-pit
+    app.get(`/html-pit` ,(req,res) => {
+        res.send(`
+        <html><body>
+            <h1>Hello</h1>
+            <img src="/images/cat.jpg" width="300" />
+        </html></body>   
+        `)
+    })
+
+//LabIndex
+//localhost:3000/labs
+    app.get(`/labs` ,(req,res) => {
+        res.send(`
+        <html>
+            <body>
+                <li>1. <a href="/profiles/joker">/profiles/joker</a></li>
+                <li>2. <a href="/api/profiles/joker">/api/profiles/joker</a></li>
+                <li>3. <a href="/cat-Simulator">/cat-simulator</a></li>
+                <li>4. <a href="/admin/grades">/admin/grades</a> AND <a href="/api/grades">/api/grades</a> 
+                AND <a href="/boredom">/boredom</a></li>
+            </body>
+        </html>
+        `)
+    })    
+
+//localhost:3000/profiles/Joker
+    app.get(`/profiles/joker`,(req,res) => {
+        res.send(`
+        <html><body>
+            <hr><h1>Mr.Kanokpol Tranfong<h1><hr>
+            <h4><br>Nickname: Joker 
+            <br>Age : 21
+            <br>Like : Computer 
+            <br>Motto: Make better not make
+            </h4>
+        </body></html>
+        `)
+    })
+
+//localhost:3000/Api/Profiles/Joker
+    app.get(`/api/profiles/joker`,(req,res) => {
+        res.send(`
+        <html><body>
+        {
+            <br>&emsp;"Name": "Kanokpol Tranfong",
+            <br>&emsp;"Nickname": "joker",
+            <br>&emsp;"Age": "21",
+            <br>&emsp;"Like": [
+                <br>&emsp;&emsp;"Computer",
+                <br>&emsp;&emsp;"Music",
+                <br>&emsp;&emsp;"Movie"
+            <br>&emsp;] ,
+            <br>&emsp;"Motto": "Make better not make"
+        <br>}    
+        </body></html>
```

Comment:
This is not an HTML route; you should be returning JSON:
```js
res.json({
  name: 'Kanokpol Tranfong',
  nickname: 'Joker',
  age: 21,
  likes: ['Computer', 'Music', 'Movie'],
  motto: 'Make better not make'
})
```

**สิ่งนี้จะไม่ส่งผลต่อเกรดของคุณ! ฉันแค่อยากจะแบ่งปันมันเพราะถ้าคุณเชี่ยวชาญแต่เนิ่นๆ มันจะช่วยคุณหลีกเลี่ยงข้อบกพร่อง**

เคล็ดลับ: เนื่องจากนี่คือ API ไม่ใช่หน้าเว็บ ฟิลด์ข้อมูลของคุณควรเป็นไปตาม "naming conventions". ภาษาโปรแกรมส่วนใหญ่มี "naming conventions".  ด้วยการตั้งชื่อคลาส ตัวแปร และวิธีการอย่างสม่ำเสมอ เราสร้างจุดบกพร่องน้อยลง และโค้ดของเรานั้นง่ายต่อการอ่านสำหรับผู้อื่น Convention ที่ได้รับความนิยมมากที่สุดคือ:

| Type | Convention | Example |
| ------------- | ------------- | ------------- |
| classes  | UpperCamelCase | class MyAwesomeClass { ... } |
| variables, fields, parameters, methods, functions  | lowerCamelCase | const myVariable = 10 |

และเรายังใช้พหูพจน์หากตัวแปรสามารถมีค่าได้หลายค่า ในรหัสของคุณ:
```
likes: ...,
motto: ...
```